src/index.js

```diff
@@ -1,41 +1,106 @@
+const Dynamic = document.querySelector('.todo-maker');
+let todos2=() => {
+    const todosArr = storage();
+    Dynamic.innerHTML = '';
+    for (let i = 0; i < todosArr.length; i += 1){
+        let isComplte = todosArr[i].completed == true ? "checked" : "";
+        Dynamic.innerHTML += `
+        <div class="list ${isComplte}">
                 <div class="list-action">
-                    <input type="checkbox">
-                    <p>${deed.description}</p>
+                    <input class ="check" id ="${todosArr[i].index}" ${isComplte} type="checkbox">
+                    <p id ="${todosArr[i].index}" contenteditable="true"class="p">${todosArr[i].description}</p>
                 </div>
                 <div class="action-container">
-                    <div><i class="fa-solid fa-ellipsis-vertical"></i></div>
+                    <div class="icon"><i id = "${todosArr[i].index}" class="fa-solid fa-ellipsis-vertical line"></i> <i id = "${todosArr[i].index}" class="fa-solid fa-trash remove"></i></div>
                 </div>
             </div>
-            <hr>`;
-            Dynamic.innerHTML += content;
-})
\ No newline at end of file
+            `;
+
+    }
+}
+
+todos2();
+
+const checked = document.querySelectorAll('.check');
+checked.forEach((button) => {
+    const todosArr = storage();
+    button.addEventListener('click', (e) => {
+       const card = e.path.filter((el) => el.classList?.contains('list')).at(0);
+       if(button.checked){
+        card.classList.add("checked");
+        todosArr[button.id].completed = true;
+       } else {
+        card.classList.remove("checked")
+        todosArr[button.id].completed = false;
+       }
+       localStorage.setItem('deeds', JSON.stringify(todosArr));
+    });
+  });
+
+document.addEventListener('keyup', (e) => {
+    const todosArr = storage();
+    if (e.key === 'Enter') {
+        localStorage.setItem('deeds', JSON.stringify(todosArr));
+         pushtodo();
+         todos2();
+         window.location.reload();
+        console.log(Dynamic)
+    }
+});
+
+const rmv = document.querySelectorAll('.remove');
+ rmv.forEach((item) => {
+  item.addEventListener('click', (e) => {
+    let todosArr = storage();
+     const card2 = e.path.filter((el) => el.classList?.contains('list')).at(0);
+     card2.remove()
+     todosArr.splice(item.id, 1);
+     todosArr.forEach((todo) => {
+        todo.index = todosArr.indexOf(todo);
+     });
+     localStorage.setItem('deeds', JSON.stringify(todosArr));
+     todos2();
+     location.reload();
+      });
+});
+
+// editing tasks
+
+const edit = document.querySelectorAll('.p');
+ edit.forEach((item) => {
+    let todosArr = storage();
+    item.addEventListener('keyup', (e) => {
+    const card2 = e.path.filter((el) => el.classList?.contains('list')).at(0);
+    let des = item.innerHTML;
+    let iD =  item.id ;
+    todosArr[iD].description = des;
+    localStorage.setItem('deeds', JSON.stringify(todosArr));
+    });
```

Comment:
- [ ] Based on the first reviewer's comments, I still can not edit a task in your To-do application. To solve this, I will advise you to do two things:

- Remove this line of code `const card2 = e.path.filter((el) => el.classList?.contains('list')).at(0);` and then
- use `textContent` instead of `innerHTML` to get the content of the `item`. 👍

Reply:
@Gambit142 I've worked on that.